app/src/processing/app/contrib/ContributionListPanel.java

```diff
@@ -78,51 +108,182 @@ public ContributionListPanel(ContributionTab contributionTab,
+      public void valueChanged(ListSelectionEvent event) {
+        //TODO this executes 2 times when clicked and 1 time when traversed using arrow keys
+        //Ideally this should always be try but while clearing the table something fishy is going on
+        if(table.getSelectedRow() != -1){
+          setSelectedPanel(panelByContribution.get(table.getValueAt(table.getSelectedRow(), 0)));
+        }
+      }
+  });
+    
+    GroupLayout layout = new GroupLayout(this);
+    layout.setHorizontalGroup(layout.createParallelGroup().addComponent(scrollPane));
+    layout.setVerticalGroup(layout.createSequentialGroup().addComponent(scrollPane));
+
+    this.setLayout(layout);
+    table.setVisible(true);
+    
   }
 
-
-  private void updatePanelOrdering() {
-    if(contributionTab.contributionType != null){
-      int row = 0;
-      for (Entry<Contribution, ContributionPanel> entry : panelByContribution.entrySet()) {
-        GridBagConstraints c = new GridBagConstraints();
-        c.fill = GridBagConstraints.HORIZONTAL;
-        c.weightx = 1;
-        c.gridx = 0;
-        c.gridy = row++;
-        c.anchor = GridBagConstraints.NORTH;
-
-        add(entry.getValue(), c);
-      }
-    } else {
-      //TODO This is where the status tab will be made
-      
-      int row = 0;
-      for (Entry<Contribution, ContributionPanel> entry : panelByContribution.entrySet()) {
-        GridBagConstraints c = new GridBagConstraints();
-        c.fill = GridBagConstraints.HORIZONTAL;
-        c.weightx = 1;
-        c.gridx = 0;
-        c.gridy = row++;
-        c.anchor = GridBagConstraints.NORTH;
-
-        add(entry.getValue(), c);
+  class StatusRendere extends DefaultTableCellRenderer {
+
+    @Override
+    public Component getTableCellRendererComponent(JTable table, Object value,
+                                                   boolean isSelected,
+                                                   boolean hasFocus, int row,
+                                                   int column) {
+      Contribution contribution = (Contribution) value;
+      JLabel label = new JLabel();
+      if (column == 0) {
+        Icon icon = null;
+        label.setBorder(BorderFactory.createEmptyBorder(0, 17, 0, 0));
+        if (contribution.isInstalled()) {
+          icon = UIManager.getIcon("OptionPane.warningIcon");
+          if (contribListing.hasUpdates(contribution)) {
+            icon = Toolkit.getLibIcon("icons/pde-16.png");
+          }
+          if (!contribution.isCompatible(Base.getRevision())) {
+            icon = Toolkit.getLibIcon("icons/pde-16.png");
+          }
+        }
+        label.setIcon(icon);
+        if (isSelected) {
+          label.setBackground(Color.BLUE);
+        }
+        label.setOpaque(true);
+//        return table.getDefaultRenderer(Icon.class).getTableCellRendererComponent(table, icon, isSelected, false, row, column);
+      } else if (column == 1) {
+        JTextPane name = new JTextPane();
```

Comment:
Can't you call `setFont()` on the `JLabel`?

How does the ellipsis implementation work in `JLabel`?